spec/betterc.dd

```diff
@@ -0,0 +1,48 @@
+Ddoc
+
+$(SPEC_S Better C,
+
+`-betterC` is a Command-Line flag for `dmd`, which generated light-weight,
+barebone, runtime-free binaries with are suitable for linking from C.
+
+---
+void main()
+{
+    import core.stdc.stdio : printf;
+    printf("Hello betterC");
+}
+---
+
+$(CONSOLE
+> dmd -betterC hello.d
+Hello betterC
+)
+
+$(H2 $(LNAME2 effects, Effects))
+
+    $(OL
+        $(LI Doesn't link DRuntime and standard library by default)
+	    $(LI Doesn't generate ModuleInfo entries)
+    )
+
+$(H2 $(LNAME2 consequences, Consequences))
+
+As no DRuntime is available, many D features won't work. For example:
+
+	$(OL
+	    $(LI No static module constructors or deconstructors)
+	    $(LI No Garbage Collection)
```

Comment:
This should be at the top, and the list should be ordered like this:
1. GC
2. Thread local storage (not sure about that)
3. TypeInfo and ModuleInfo
3. extern(D) classes, because of 3) and in part because of 1) (the new expression, though there are various ways around that).
4. No built-in threading (a consequence of 4), i.e. no `core.thread`, `std.concurrency`, `std.parallelism`, vibe.d, etc.
4. Dynamic arrays (but not slices) and associative arrays
5. Exceptions
6. `switch` with strings
7. `final switch` and other `assert`-like features like bounds checking (@WalterBright please correct me, if I'm wrong)
7. `synchronized`
8. For reference on all (I think) the runtime hooks, see: https://github.com/dlang/dmd/blob/master/src/ddmd/backend/rtlsym.h